landoapi/hgexports.py

```diff
@@ -605,3 +616,144 @@ def run_diff_checks(self) -> list[str]:
+        commit_message = patch_helper.get_commit_description()
+
+        # Skip the check if the `skip_check` flag is set.
+        if self.skip_check or "SKIP_BMO_CHECK" in commit_message:
+            self.skip_check = True
+            return
+
+        self.bug_ids |= set(parse_bugs(commit_message))
+
+    def result(self) -> Optional[str]:
+        if self.skip_check or not self.bug_ids:
+            return
+
+        try:
+            found_bugs = search_bugs(self.bug_ids)
+        except requests.exceptions.RequestException as exc:
+            return (
+                "Could not contact BMO to check for security bugs referenced in commit message. "
+                f"Use `SKIP_BMO_CHECK` in your commit message to push anyways. Error: {str(exc)}."
+            )
+
+        invalid_bugs = self.bug_ids - found_bugs
+        if not invalid_bugs:
+            return
+
+        # Check a single bug only.
+        bug_id = invalid_bugs.pop()
+        try:
+            status_code = get_status_code_for_bug(bug_id)
+        except requests.exceptions.RequestException as exc:
+            return (
+                "Could not contact BMO to check for security bugs referenced in commit message. "
+                f"Use `SKIP_BMO_CHECK` in your commit message to push anyways. Error: {str(exc)}."
+            )
+
+        if status_code == 401:
+            return (
+                f"Your commit message references bug {bug_id}, which is currently private. To avoid "
+                "disclosing the nature of this bug publicly, please remove the affected bug ID "
+                "from the commit message."
+            )
+
+        if status_code == 404:
+            return (
+                f"Your commit message references bug {bug_id}, which does not exist. Please check your "
+                "commit message and try again."
+            )
+
+        return (
+            f"While checking if bug {bug_id} in your commit message is a security bug, an error occurred "
+            "and the bug could not be verified."
+        )
+
+
+@dataclass
+class PushAssessor:
+    """Assess pushes for landing issues."""
+
+    patch_helpers: Iterable[PatchHelper]
+    repo: Repo
+
+    def run_push_checks(
+        self, push_checks: Optional[list[Type[PushCheck]]] = None
+    ) -> list[str]:
+        """Execute the set of checks on the diffs, returning a list of issues.
+
+        `push_checks` specifies the push-wide checks to run on the push, otherwise
+        all checks will be run.
+        """
+        issues = []
+
+        if not push_checks:
+            push_checks = [BugReferencesCheck]
+
+        checks = [
+            check(repo=self.repo)
+            for check in push_checks
+            if check(repo=self.repo).relevant()
+        ]
+
+        for patch_helper in self.patch_helpers:
+            # Pass the patch information into the push-wide check.
+            for check in checks:
+                check.next_diff(patch_helper)
```

Comment:
This is a little confusing here. If the purpose of this call is to set `check.skip_check` and `check.bug_ids`, maybe there's a clearer way of doing that elsewhere more explicitly.

Reply:
This is where all the push checks are passed each individual diff. Setting `skip_check` and `bug_ids` will only happen for the `BugReferencesCheck`, and other push-wide checks will have different behaviour.